x/subspaces/client/cli/tx.go

```diff
@@ -0,0 +1,382 @@
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// GetCmdEditSubspace returns the command to edit a subspace
+func GetCmdEditSubspace() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "edit [subspace-id]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Edit the subspace with the given id",
+		Example: fmt.Sprintf(`
+%s tx subspaces edit 1 \
+  --name "Desmos - Democratizing social networks"
+  --description "The official subspace of Desmos" \
+  --treasury desmos1jqk5p244yl4ktukq5xhavvlfzl8z4we4qfmuyh \
+  --owner desmos1p8r4guvdze03md4g9zclhh6mr8ljvtd80pehr3 \
+  --from alice
+`, version.AppName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			subspaceID, err := types.ParseSubspaceID(args[0])
+			if err != nil {
+				return err
+			}
+
+			name, err := cmd.Flags().GetString(FlagName)
+			if err != nil {
+				return err
+			}
+
+			description, err := cmd.Flags().GetString(FlagDescription)
+			if err != nil {
+				return err
+			}
+
+			treasury, err := cmd.Flags().GetString(FlagTreasury)
+			if err != nil {
+				return err
+			}
+
+			owner, err := cmd.Flags().GetString(FlagOwner)
+			if err != nil {
+				return err
+			}
+
+			msg := types.NewMsgEditSubspace(subspaceID, name, description, treasury, owner, clientCtx.FromAddress.String())
+			if err = msg.ValidateBasic(); err != nil {
+				return fmt.Errorf("message validation failed: %w", err)
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	cmd.Flags().String(FlagName, types.DoNotModify, "New human readable name of the subspace")
+	cmd.Flags().String(FlagDescription, types.DoNotModify, "Description of the subspace")
+	cmd.Flags().String(FlagTreasury, types.DoNotModify, "Treasury of the subspace")
+	cmd.Flags().String(FlagOwner, types.DoNotModify, "Owner of the subspace")
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// GetCmdCreateUserGroup returns the command to create a user group
+func GetCmdCreateUserGroup() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create-user-group [subspace-id] [group-name] [[permissions]]",
+		Args:  cobra.MinimumNArgs(2),
+		Short: "Create a new user group within a subspace",
+		Long: `Create a new user group within the subspace having the provided id.
+The permissions this group can be set using the third (optional) parameter.
+If no permissions are set, the default PermissionNothing will be used instead.
+Multiple permissions must be specified separating them with a comma (,).`,
+		Example: fmt.Sprintf(`
+%s tx subspaces create-user-group 1 "Admins" "Write,ModerateContent,SetPermissions" \
+  --from alice
+`, version.AppName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			subspaceID, err := types.ParseSubspaceID(args[0])
+			if err != nil {
+				return err
+			}
+
+			name := args[1]
```

Comment:
I think the name of the group should be saved as case insensitive so that we can ensure that in a subspace there won't be two groups named for example: "Admins" and "admins".
To do it, you can simply use:
```go
strings.ToLower(args[1])
```

Reply:
You're right. I think it might be even better to create a `UserGroup` type as follows: 

```proto 
message UserGroup {
  uint32 id = 1; 
  string name = 2; 
  string description = 3
}
```

By doing so we can refer to different groups using their IDs rather than the name. We can assume that the subspace admin won't create multiple groups with different names. And even if they do, they can simply delete them later. 
But we could have an extensible type for the future if we want to add more things to it. What do you think (also @dadamu)?

Reply:
Yes I think `UserGroup`s deserves their own struct. We will be able to handle them better later on.

Reply:
Agree with we should have `UserGroup` struct. But if we use group_id instead, it breaks the `SetPermissions` message and permission store key, the type of target is string now.

Reply:
All right, will implement the `UserGroup` type. I think we can have the group permissions as a field of the `UserGroup`, and then different messages for users/groups separately (eg. `MsgSetUserPermission` and `MsgSetGroupPermission`). What do you think @dadamu?

Reply:
@RiccardoM It is good to me.